Using feature overrides to make changes, and general layer cleanup

ahsanhussain opened the pull request.

kergoth commented:
These commits are quite monolithic compared to how we usually do things. I find if you're making bullets in your commit message, it might be time to break things up. Any thoughts on this, @abelal?

ahsanhussain commented:
@kergoth I've reasonably split the changes. Let me know if this works. Awais will be busy, and won't be able to take a look before Monday but we have the sprint closing today.

kergoth commented on meta-sokol-flex-staging/dynamic-layers/openembedded-layer/recipes-support/pidgin/pidgin_%.bbappend:
```diff
@@ -2,4 +2,4 @@
 # ---------------------------------------------------------------------------------------------------------------------
 
-RDEPENDS:finch += "python3-core"
+RDEPENDS:finch:feature-sokol-flex-staging += "python3-core"
```
This should be `:append:feature-sokol-flex-staging` to mirror existing behavior, this version is entirely replacing `RDEPENDS:finch` rather than appending to it,  unless there's a reason we're changing the behavior here.

kergoth commented on meta-sokol-flex-staging/dynamic-layers/virtualization-layer/recipes-containers/docker-compose/python3-docker-compose_%.bbappend:
```diff
@@ -2,4 +2,4 @@
 # ---------------------------------------------------------------------------------------------------------------------
 
-RDEPENDS:${PN} += "${PYTHON_PN}-distutils"
+RDEPENDS:${PN}:feature-sokol-flex-staging += "${PYTHON_PN}-distutils"
```
Same issue as finch, append is being converted to appending to an override which replaces the original rather than appending to it.

kergoth commented on meta-sokol-flex-staging/security/recipes-ids/tripwire/tripwire_%.bbappend:
```diff
@@ -3,7 +3,7 @@
 # Use vi by default to avoid pulling in nano
-RDEPENDS_REMOVE = "nano"
+RDEPENDS_REMOVE:feature-sokol-flex-staging = "nano"
 RDEPENDS:${PN}:remove:feature-sokol-flex-staging = " ${RDEPENDS_REMOVE}"
```
This isn't needed, we already use this override here, right below that line, where we actually use this variable.

ahsanhussain replied:
Must have been doing this late at night. * smh *

kergoth commented on meta-sokol-flex-distro/recipes-core/initrdscripts/initramfs-framework_%.bbappend:
```diff
@@ -2,7 +2,7 @@
-RDEPENDS:initramfs-module-lvm += "lvm2"
-RRECOMMENDS:${PN}-base += "initramfs-module-lvm"
+RDEPENDS:initramfs-module-lvm:sokol-flex += "lvm2"
+RRECOMMENDS:${PN}-base:sokol-flex += "initramfs-module-lvm"
```
Same issue as finch, append is being converted to appending to an override which replaces the original rather than appending to it.

kergoth commented on meta-sokol-flex-common/classes/mirrors.bbclass:
```diff
@@ -2,11 +2,11 @@
 "
 
-MIRRORS += "\
+MIRRORS:sokol-flex += "\
```
Same issue as finch, append is being converted to appending to an override which replaces the original rather than appending to it.

kergoth commented:
To clarify, I think the concept is sound, I'd say there's no need to keep the RFC tag, my only concerns are minor technical issues, you've done well with this.

ahsanhussain commented:
@kergoth I fixed these finch issue instances for some recipes, after I got do_configure() failure with ccid and one other package. Looks like these ones got missed, as the build went through. Should have taken a look at the final diff again. Fixed now

kergoth commented on meta-sokol-flex-common/classes/mirrors.bbclass:
```diff
@@ -2,11 +2,11 @@
 "
 
-MIRRORS += "\
+MIRRORS:append:sokol-flex = "\
```
`:append` without a leading space. Sorry :)

ahsanhussain replied:
Deliberately done; does it take white space or newline here?
